cli/src/semgrep/semgrep_main.py

```diff
@@ -172,27 +171,42 @@ def run_rules(
             output_handler.handle_semgrep_errors(join_rule_errors)
 
     if len(dependency_aware_rules) > 0:
-        from semgrep.dependency_aware_rule import run_dependency_aware_rule
-        from semdep.find_lockfiles import make_dependency_trie
-
-        targets = [t.path for t in target_manager.targets]
-        top_level_target_rooted = list(targets[0].parents)
-        top_level_target: Path = (
-            targets[0]
-            if len(top_level_target_rooted) == 0
-            else top_level_target_rooted[-1]
+        from semgrep.dependency_aware_rule import (
+            generate_non_reachable_sca_findings,
+            generate_reachable_sca_findings,
         )
-        namespaces = list({ns for r in dependency_aware_rules for ns in r.namespaces})
-        dep_trie = make_dependency_trie(top_level_target, namespaces, target_manager)
 
         for rule in dependency_aware_rules:
-            (dep_rule_matches, dep_rule_errors,) = run_dependency_aware_rule(
-                rule_matches_by_rule.get(rule, []),
-                rule,
-                dep_trie,
-            )
-            rule_matches_by_rule[rule] = dep_rule_matches
-            output_handler.handle_semgrep_errors(dep_rule_errors)
+            if rule.should_run_on_semgrep_core:
+                (
+                    dep_rule_matches,
+                    dep_rule_errors,
+                    matched_lockfiles,
+                ) = generate_reachable_sca_findings(
+                    rule_matches_by_rule.get(rule, []),
+                    rule,
+                )
+                rule_matches_by_rule[rule] = dep_rule_matches
+                output_handler.handle_semgrep_errors(dep_rule_errors)
+                (
+                    dep_rule_matches,
+                    dep_rule_errors,
+                    targeted_lockfiles,
+                ) = generate_non_reachable_sca_findings(
+                    rule, target_manager, matched_lockfiles
+                )
+                rule_matches_by_rule[rule].extend(dep_rule_matches)
+                output_handler.handle_semgrep_errors(dep_rule_errors)
+                all_targets.union(targeted_lockfiles)
```

Comment:
Do you also need to union with `matched_lockfiles`?

Reply:
Maybe... If we scan a lockfile as part of reachable result, but the lockfile itself was not a target of the scan, should we include it in the scanned files?

Reply:
oh I see yes. I think makes sense to not include it